### PR TITLE
SSO Authorization process

### PR DESCRIPTION
**Pull request by DanielePalombo**
This PR adds [omniauth-bolt](https://github.com/nebulab/omniauth-bolt) strategy

Screenshot 2022-05-09 at 21 35 09

https://github.com/nebulab/solidus_bolt/issues/18

### CODE REVIEW

File: app/models/solidus_bolt/bolt_configuration.rb [hunk 1]
```diff
@@ -34,6 +34,14 @@ def environment_url
       end
     end
 
+    def embed_js
+      case BoltConfiguration.environments[environment]
+      when 0 then 'https://connect.bolt.com/embed.js'
+      when 1 then 'https://connect-sandbox.bolt.com/embed.js'
+      when 2 then 'https://connect-staging.bolt.com/embed.js'
```
Comment (DanielePalombo) [inline app/models/solidus_bolt/bolt_configuration.rb:41]: We don't have the staging option. https://docs.bolt.com/developers/guides/checkout-guides/embedded-checkout/accounts/setup/#embedjs

Comment (DanielePalombo) [reply app/models/solidus_bolt/bolt_configuration.rb]: https://github.com/nebulab/solidus_bolt/commit/971f2cfda14aa0e4856498e7fc576655cbe06583

**Comment by DanielePalombo** [comment]
I have fixed the spec error with [this commit](https://github.com/nebulab/solidus_bolt/commit/58e0faa925289c44e6d73ff187e72a5b0369182f)

**Comment by DanielePalombo** [comment]
I tried to investigate the failing spec, but without success. I think the problem is related to docker and circleCI orb image. I'm moving this PR to `Ready for review` and I think we can merge it if everything is working properly on our env.

File: bin/sandbox [hunk 2]
```diff
@@ -51,6 +51,8 @@ cd ./sandbox
 cat <<RUBY >> Gemfile
 gem 'solidus', github: 'solidusio/solidus', branch: '$BRANCH'
 gem 'solidus_auth_devise', '>= 2.1.0'
+gem 'omniauth-bolt', github: 'nebulab/omniauth-bolt'
```
Comment (Naokimi) [inline bin/sandbox:54]: you are adding the gem here but not in the dependencies - could be the reason why the tests are failing

**Review by Naokimi — APPROVED** [review]

File: app/models/solidus_bolt/bolt_configuration.rb [hunk 3]
```diff
@@ -34,6 +34,14 @@ def environment_url
       end
     end
 
+    def embed_js
+      if BoltConfiguration.environments[environment] == 0
```
[the remark this answers is not in the document]
Comment (DanielePalombo) [reply app/models/solidus_bolt/bolt_configuration.rb]: https://github.com/nebulab/solidus_bolt/commit/5814af4b953097928aa1d24024d85ebcc97063bd

**Review by MassimilianoLattanzio — APPROVED** [review]
LGTM

**Pull request merged**